src/Vlingo.Common/Identity/UniqueTextGenerator.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) 2012-2019 Vaughn Vernon. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the
+// Mozilla Public License, v. 2.0. If a copy of the MPL
+// was not distributed with this file, You can obtain
+// one at https://mozilla.org/MPL/2.0/.
+
+using System;
+using System.Text;
+
+namespace Vlingo.Common.Identity
+{
+    /// <summary>
+    /// Generates unique text. The longer the generated String the more likely
+    /// it is to be unique. In tests, a length-10 String is consistently unique
+    /// to one million instances and greater (actually 100 million but requires
+    /// several seconds). When using length-7, -8, and -9, the uniqueness fails
+    /// before one million. Thus, the default length is 10. If you intend to use
+    /// this generator for short, unique identities, you should plan to ensure
+    /// uniqueness. Although the likelihood of uniqueness is high, you should
+    /// still compare against previously generated values, and if non-unique,
+    /// throw it out and retry generation. The uniqueness has not been tested
+    /// across simultaneously running machines.
+    /// </summary>
+    public class UniqueTextGenerator
```

Comment:
I don't know if we need that, but should we use `salt`  for predictible output in UT or while testing ?

Reply:
True. It can be made more predictable. But, it would require changes in the `Random` initialization in the class. And, the salt needs be injectable or needs to come from "some" configuration somewhere. That's why I left it as the Java version. On the other, the default Random class from .NET is same as vlingo-net-uuid's `RandomBasedGenerator`. We will get "pretty good" randomness by default, just not the predictable behavior for Unit Tests.

Reply:
Perhaps I don't understand `RandomBasedGenerator` fully, but if it returns a `Guid` then the minimum length is 16 bytes in natural form and 32 in string form (and may be larger since the default string is 36 characters). The point of this generator is to produce the shortest possible random-based unique strings. The shortest string that I have consistently generated in Java over many millions of ids is length 10. You will see uniqueness with shorter strings, such as 7, but only over tens of thousands of ids.

BTW, if you see a problem in the Java version with the randomness please indicate so I can improve the generation.